eox_tagging/api/v1/serializers.py

```diff
@@ -0,0 +1,84 @@
+"""
+Serializers for tags and related objects.
+"""
+import re
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext as _
+from rest_framework import serializers
+
+from eox_tagging.api.v1 import fields
+from eox_tagging.constants import AccessLevel, Status
+from eox_tagging.edxapp_wrappers import get_object, get_site
+from eox_tagging.models import Tag
+
+PROXY_MODEL_NAME = "opaquekeyproxymodel"
+MODELS_WITH_COMPOUND_KEYS = {
+    "courseenrollment": ["username", "course_id"]  # Compound keys
+}
+
+
+class TagSerializer(serializers.ModelSerializer):
+    """Serializer for tag objects."""
+    target_id = serializers.CharField(source='target_object')
+    owner_id = serializers.CharField(source='owner_object', required=False)
+    owner_type = serializers.CharField(source='owner_object_type')
+    target_type = serializers.CharField(source='target_object_type')
+    access = fields.EnumField(enum=AccessLevel)
+    status = fields.EnumField(enum=Status, required=False)
+
+    class Meta:  # pylint: disable=old-style-class, useless-suppression
+        """Meta class."""
+        model = Tag
+        fields = ('key', 'tag_value', 'tag_type', 'access', 'activation_date', 'expiration_date',
+                  'target_id', 'owner_id', 'owner_type', 'target_type', 'status')
+
+    # Validation and creation of tags
+    def create(self, validated_data):
+        """Function that creates a Tag instance."""
+
+        # Finding target and owner objects
+        target_object = None
+        owner_object = None
```

Comment:
This is not necessary

Reply:
I thought so, but when I was testing using the platform it gave me this error: reference before assignment. After I added those lines, the error was gone

Reply:
That shouldn't happen if you apply the suggestion of the line 60